src/org/labkey/test/tests/issues/IssueAPITest.java

```diff
@@ -0,0 +1,377 @@
+        IssueModel originalIssue = basicIssueModel(title, "Gonna assign this");
+        IssueModel updateIssue = basicIssueModel(null, "assigned now")
+                .setAction(IssueModel.IssueAction.update)
+                .setAssignedTo(TEST_BUDDY_ID);
+
+        // create the issue
+        var issueId = doIssueAction(originalIssue);
+
+        // assign the issue
+        updateIssue.setIssueId(issueId);
+        doIssueAction(updateIssue);
+
+        // get the updated result
+        var updatedIssue = getIssueResponse(issueId);
+        assertEquals("should be assigned to buddy", TEST_BUDDY_ID, updatedIssue.getAssignedTo());
+        assertEquals("passing null title should not update", title, updatedIssue.getTitle());
+    }
+
+    @Test
+    public void testReopenAnIssue() throws Exception
+    {
+        IssueModel originalIssue = basicIssueModel("Reopen test issue", "Gonna close and reopen this");
+        IssueModel closeIssue = basicIssueModel(null, null)
+                .setAction(IssueModel.IssueAction.close);
+        IssueModel reopenIssue = basicIssueModel(null, null)
+                .setAction(IssueModel.IssueAction.reopen);
+
+        // insert
+        var issueId = doIssueAction(originalIssue);
+
+        // close
+        closeIssue.setIssueId(issueId);
+        doIssueAction(closeIssue);
+
+        // verify
+        var closedIssue = getIssueResponse(issueId);
+        assertEquals("closed", closedIssue.getStatus());
+
+        // reopen
+        reopenIssue.setIssueId(issueId);
+        doIssueAction(reopenIssue);
+
+        // verify
+        var updatedIssue = getIssueResponse(issueId);
+        assertEquals("open", updatedIssue.getStatus());
+    }
+
+    @Test
+    public void testUpdateMultipleIssues() throws Exception
+    {
+        File firstFile = TestFileUtils.writeTempFile("firstErr.txt", "first error of 2");
+        File secondFile = TestFileUtils.writeTempFile("secondErr.txt", "second error of 2");
+        IssueModel firstIssue = basicIssueModel("first of three", "to test multiple issues at once")
+                .addAttachment(firstFile);
+        IssueModel secondIssue = basicIssueModel("second of three", "to test multiple issues at once")
+                .addAttachment(secondFile);
+        IssueModel thirdIssue = basicIssueModel("third of three", "to test multiple issues at once")
+                .addAttachment(firstFile).addAttachment(secondFile);
+        List<IssueModel> issues = new ArrayList<>();
+        issues.add(firstIssue);
+        issues.add(secondIssue);
+        issues.add(thirdIssue);
+
+        IssuesCommand insertCmd = new IssuesCommand();
+        insertCmd.setIssues(issues);
+        var insertResponse = insertCmd.execute(createDefaultConnection(), getProjectName());
+        List<Long> issueIds =  insertResponse.getIssueIds();
+        for (Long issueId : issueIds)
+        {
+            var issueresponseModel = getIssueResponse(issueId);
```

Comment:
Blindly validating in a loop like this is **BAD**!
The insert command might return nothing, and you validate nothing.
It might return the same issue ID three times, or a dozen times, and you just validate the same repeatedly.

I bet you this command is only inserting a single issue due to a bug in the `IssuesCommand` and the test is totally missing it.

Reply:
Well, when I debugged this, it hit every case.  I can add validation that there were 3 issues in the response to catch the empty response case tho

Reply:
Well, this was a good catch- it's returning the same IssueId, 3 times.  I've added a check for distinctness of returned Ids that does catch it.  Now the question is how to actually send all 3 issues

Reply:
It turns out the bug was in the deserializing- the API was sending back 3 IDs, IssueResponse was writing down the first one 3 times (again, need the facepalm emoji something fierce here)